public/app/plugins/panel/graph/specs/graph_tooltip.test.ts

```diff
@@ -97,42 +84,34 @@ describeSharedTooltip('steppedLine false, stack false', (ctx: any) => {
         hideTooltip: false,
       },
     ];
-    ctx.pos = { x: 11 };
-  });
+    const pos = { x: 11 };
 
-  it('should return 2 series', () => {
-    expect(ctx.results.length).toBe(2);
-  });
-
-  it('should add time to results array', () => {
-    expect(ctx.results.time).toBe(10);
-  });
+    it('should return 2 series', () => {
+      // @ts-ignore
+      const tooltip = new GraphTooltip(elem, dashboard, scope, getSeriesFn);
+      const results = tooltip.getMultiSeriesPlotHoverInfo(data, pos);
+      expect(results.length).toBe(2);
+    });
 
-  it('should set value and hoverIndex', () => {
-    expect(ctx.results[0].value).toBe(15);
-    expect(ctx.results[1].value).toBe(2);
-    expect(ctx.results[0].hoverIndex).toBe(0);
-  });
-});
+    it('should add time to results array', () => {
+      // @ts-ignore
+      const tooltip = new GraphTooltip(elem, dashboard, scope, getSeriesFn);
+      const results = tooltip.getMultiSeriesPlotHoverInfo(data, pos);
+      expect(results.time).toBe(10);
+    });
 
-describeSharedTooltip('one series is hidden', (ctx: any) => {
-  ctx.setup(() => {
-    ctx.data = [
-      {
-        data: [
-          [10, 15],
-          [12, 20],
-        ],
-      },
-      { data: [] },
-    ];
-    ctx.pos = { x: 11 };
+    it('should set value and hoverIndex', () => {
+      // @ts-ignore
```

Comment:
just curious why we need to ignore here? maybe we could write a comment on why in case we can remove it in the future

Reply:
i have no idea - it's simply copied from here: https://github.com/grafana/grafana/pull/42681/files/4909655b0a36ba8a10e3023a84a9f1e483c62ba3#diff-0ec0212a45874dace81fd1cae5d23882b7fa58567fdad3d5d31f03f6f7321678L34 since there isn't 1 function that initialises the component anymore

Reply:
I think it's because `GraphTooltip` is a function and we're calling it with `new` keyword but TS expects it to be a class to properly infer the types.